rust-semantics/preprocessing/initialization.md

```diff
@@ -7,6 +7,56 @@ module INITIALIZATION
     imports private RUST-PREPROCESSING-PRIVATE-HELPERS
     imports private RUST-PREPROCESSING-PRIVATE-SYNTAX
 
+    rule
+        <k> structInitializer( (struct Name:Identifier { F:StructFields } ):Struct, ParentPath:TypePath) => structParser(append(ParentPath, Name), F:StructFields )
+            ...
+        </k>
+        <struct-list> .List => ListItem(append(ParentPath, Name)) ...</struct-list>
+        <structs>
+            ...
+            .Bag
+            =>  <struct>
+                    <struct-path> append(ParentPath, Name) </struct-path>
+                    ...
+                </struct>
+        </structs>
+        
+    rule
+        <k> structParser(Name:TypePath, ((FN:Identifier : FT:Type):StructField  , RF:StructFields):StructFields) => 
+                structParser(Name:TypePath, RF:StructFields)
+            ...
+        </k>
+        <struct>
+          ...
+          <struct-path> Name </struct-path>
+          <variable-list> L:List => ListItem(FN) L </variable-list>
+          <variables>
+            .Bag =>
+              <variable>
+                <variable-name> FN </variable-name>
+                <variable-type> FT </variable-type>
+              </variable>
+            ...
+          </variables>
+        </struct> [owise]
```

Comment:
Why is this an owise rule?

Reply:
Testing it without the `[owise]` always caused tests to reach a stuck state. My understanding is that there is a certain degree of ambiguity between this rule and the rule right below it, and when the second one should be applied to finish the parsing of the struct declaration, it would apply the first, leading the K cell to a state where it could not be rewritten further.

Reply:
Ah, now I understand what happens... I was misreading the rule below. You may try to replace it with
```
rule structParser(_Name:TypePath, .StructFields) => .K
```
and then you will probably not need an owise rule here.